src/adagrams.js

```diff
@@ -1,15 +1,95 @@
+const letterDistribution = {
+  A: 9, B: 2, C: 2, D: 4, E: 12, F: 2, G: 3, H: 2, I: 9, J: 1, K: 1,
+  L: 4, M: 2, N: 6, O: 8, P: 2, Q: 1, R: 6, S: 4, T: 6, U: 4, V: 2, W: 2, X: 1,
+  Y: 2, Z: 1
+};
+
+const letterScores = {
+  A: 1, E: 1, I: 1, O: 1, U: 1, L: 1, N: 1, R: 1, S: 1, T: 1,
+  D: 2, G: 2,
+  B: 3, C: 3, M: 3, P: 3,
+  F: 4, H: 4, V: 4, W: 4, Y: 4,
+  K: 5,
+  J: 8, X: 8,
+  Q: 10, Z: 10
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const availableLetters = Object.keys(letterDistribution);
+  const hand = [];
+
+  for (let count = 0; count < 10; count++) {
+    const randomIndex = Math.floor(Math.random() * availableLetters.length);
+    const selectedLetter = availableLetters[randomIndex];
+
+    if (letterDistribution[selectedLetter] > 0) {
+      hand.push(selectedLetter);
+      letterDistribution[selectedLetter] -= 1;
+    }
+    if (letterDistribution[selectedLetter] === 0) {
+      availableLetters.splice(randomIndex, 1);
+    }
+  }
+
+  return hand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const inputWord = input.toUpperCase();
+  const inputLetters = inputWord.split('');
+  const letterCounts = {};
+  
+  for (let letter of lettersInHand) {
+    letterCounts[letter] = letterCounts[letter] ? letterCounts[letter] + 1 : 1;
+  }
+
+  for (let letter of inputLetters) {
+    if (!(letter in letterCounts) || letterCounts[letter] === 0) {
+      return false;
+    }
+    letterCounts[letter]--;
+  }
+
+  return true;
 };
```

Comment:
While this works, it starts to get a little overly complicated with multiple for loops as we build an extra object for the count of each letter. One alternative path here we could do is make a shallow copy of lettersInHand. From there, we could check each letter in input. If it's in the copy, we remove it, if not, we return false.